cms/static/js/spec/views/pages/container_spec.js

```diff
@@ -591,6 +591,47 @@ define(["jquery", "underscore", "underscore.string", "js/common_helpers/ajax_hel
                         });
                     });
                 });
+
+                if (global_page_options.enable_paging) {
+                    describe("Previews", function(){
+
+                        var getButtonIcon, getButtonText;
+
+                        getButtonIcon = function(containerPage){
+                            return containerPage.$('.action-toggle-preview .preview-arrow');
+                        };
+
+                        getButtonText = function(containerPage) {
+                            return containerPage.$('.action-toggle-preview .preview-text').text().trim();
+                        };
+
+                        it('initially shows loading icon and reads "Loading..."', function(){
+                            containerPage = getContainerPage();
+                            expect(getButtonIcon(containerPage)).toHaveClass('icon-refresh');
+                            expect(getButtonText(containerPage)).toBe("${_('Loading...')}");
+                        });
+
+                        function updatePreviewButtonTest(label, show_previews, expected_icon_class, expected_text){
+                            it('can set preview button to '+label, function (){
+                                containerPage = getContainerPage();
+                                containerPage.updatePreviewButton(show_previews);
+                                expect(getButtonIcon(containerPage)).toHaveClass(expected_icon_class);
+                                expect(getButtonText(containerPage)).toBe(expected_text);
+                            });
+                        }
+                        updatePreviewButtonTest('show previews', true, 'icon-arrow-up', 'Hide Previews');
```

Comment:
These aren't the icons that are shown. On sandbox I see a recycle icon that is the same for both Hide Previews and Show Previews.

Nit: I would simplify the args to the test case by getting rid of "label" (just use expected_text).